src/Gui/App.luau

```diff
@@ -0,0 +1,55 @@
+local Hoarcekat = script:FindFirstAncestor("Hoarcekat")
+local ReactRoblox = require(Hoarcekat.Packages.ReactRoblox)
+local React = require(Hoarcekat.Packages.React)
+
+local e = React.createElement
+local MainContext = require(Hoarcekat.Plugin.Gui.Contexts.MainContext)
+local StudioTheme = require(Hoarcekat.Plugin.Gui.Contexts.StudioTheme)
+local VerticalSplitter = require(Hoarcekat.Plugin.Gui.Components.VerticalSplitter)
+local Sidebar = require(Hoarcekat.Plugin.Gui.Components.Sidebar)
+local Preview = require(Hoarcekat.Plugin.Gui.Components.Preview)
+
+type Props = {
+	root: ReactRoblox.RootType,
+	plugin: any,
+	children: any?,
+	mouse: PluginMouse,
+	savedState: { selectedStory: Instance? }?,
+}
+
+local function App(props: Props)
+	local theme = React.useContext(StudioTheme.Context)
+
+	return e(StudioTheme.Provider, nil, {
+		MainContext = e(MainContext.Provider, {
```

Comment:
For context providers, or anything else that is "transparent", you should prefer to use the node directly:

```lua
return e(ProviderA, {}, e(ProviderB, {}, {
    -- everything else
})
```

The names are not used anywhere, so we don't want to pollute with them